Modules/OpenGhostDir.cs

```diff
@@ -10,50 +10,146 @@ internal class OpenGhostDir : Module
 
 
-        public static string GetGhostDirectory()
+        public static string GetGhostDirectory(LevelData level = null)
         {
-            string path = GhostRecorder.GetCompressedSavePathForLevel(Singleton<Game>.Instance.GetCurrentLevel());
+            if (level == null)
+                level = Singleton<Game>.Instance.GetCurrentLevel();
+            string path = GhostRecorder.GetCompressedSavePathForLevel(level);
             path = Path.GetDirectoryName(path) + "/";
             return path;
         }
 
+        private static void SetupButton(MenuButtonHolder button, LevelData level = null)
+        {
+            try
+            {
+                if (ghostButton != null)
+                {
+                    if (jobArchive)
+                    {
+                        UnityEngine.Object.Destroy(ghostButton.gameObject);
+                        UnityEngine.Object.Destroy(ghostButton.transform.parent.gameObject);
+                        ghostButton = null;
+                    }
+                }
+
+                if (ghostButton == null)
+                {
+                    // copy the layout
+                    var layout = UnityEngine.Object.Instantiate(button.transform.parent.gameObject, button.transform.parent.parent);
+                    layout.name = "Ghost Button Holder";
+                    // empty it
+                    foreach (Transform child in layout.transform)
+                        UnityEngine.Object.Destroy(child.gameObject);
+
+                    var ghostObject = UnityEngine.Object.Instantiate(button.gameObject, layout.transform);
+                    ghostButton = ghostObject.GetComponent<MenuButtonHolder>();
+                    var backComponent = ghostObject.GetComponent<BackButtonAccessor>();
+                    if (backComponent)
+                        UnityEngine.Object.Destroy(backComponent);
+
+                    ghostButton.ButtonRef.onClick.RemoveAllListeners();
+                    ghostButton.ButtonRef.onClick.AddListener(() => Process.Start("file://" + GetGhostDirectory(level)));
+                }
+
+                ghostButton.name = "Button Ghost";
+                ghostButton.buttonText = "Open Ghost Directory";
+                ghostButton.buttonTextRef.text = "Open Ghost Directory";
+                ghostButton.gameObject.SetActive(_setting_GhostButton.Value);
+            }
+            catch (Exception e)
+            {
+                UnityEngine.Debug.LogError("error on setting up the button " + e);
+            }
+        }
+
         [HarmonyPrefix]
         [HarmonyPatch(typeof(MenuScreenResults), "OnSetVisible")]
         private static void OnMenuScreen(ref MenuScreenResults __instance)
         {
-            if (ghostButton == null)
+            try
```

Comment:
Try to put only what's needed in the try block. This improves readability

Reply:
Same as above.